parser/dns.go

```diff
@@ -6,20 +6,43 @@ import (
 	"github.com/activecm/rita/parser/parsetypes"
 	"github.com/activecm/rita/pkg/data"
 	"github.com/activecm/rita/pkg/hostname"
+
+	log "github.com/sirupsen/logrus"
 )
 
-func parseDNSEntry(parseDNS *parsetypes.DNS, filter filter, retVals ParseResults) {
+func parseDNSEntry(parseDNS *parsetypes.DNS, filter filter, retVals ParseResults, logger *log.Logger) {
 
 	// extract and store the dns client ip address
 	src := parseDNS.Source
 	srcIP := net.ParseIP(src)
 
+	// verify that ip address was parsed successfully
+	if srcIP == nil {
+		logger.WithFields(log.Fields{
+			"uid": parseDNS.UID,
+			"src": parseDNS.Source,
+		}).Error("Unable to get valid client ip address from dns log entry, skipping entry.")
+		return
+	}
+
+	// get domain
+	domain := parseDNS.Query
+
+	// verify that domain was parsed successfully
+	if domain == "" {
```

Comment:
The domain may be empty in valid DNS queries. 
Here is an example entry in a Zeek log which has an `(empty)` query field:
```
1518765993.164098       CQrwEE4MAKPCM5ATGd      192.168.88.2    12058   199.7.83.42     53      udp     2409    0.073556        (empty) 1       C_INTERNET      48      DNSKEY  0       NOERROR T       F       F       F       1       <unknown type=48>,<unknown type=48>,<unknown type=48>,<unknown type=46> 172800.000000,172800.000000,172800.000000,172800.000000 F
```

Technically this change will throw out some data RITA would previously accept. If a client makes a dummy DNS A type query for "", currently, we would add that client's IP address to the hostname collection record for "". This check would prevent a hostname record being created for "". 

I lean towards removing this change to keep the current behavior. We may want to reconsider how we handle DNS records with an empty query, but I think it would be better addressed as a separate issue.